bootloader/src/old_machine_bootloader_linux.c

```diff
@@ -0,0 +1,312 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <curl/curl.h>
+#include <stdbool.h>
+#include <wchar.h>
+#include <sys/utsname.h>
+#include <unistd.h>
+#include <gnu/libc-version.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+
+#define minVersion 6.1
+
+struct response {
+  char *ptr;
+  size_t len;
+};
+
+struct requestData {
+    char* IPstring;
+    char* glibcVersion;
+    char* hostname;
+    char* osVersion;
+    bool tunnelUsed;
+    char* tunnel;
+};
+
+char* getRequestDataJson(struct requestData reqData){
+    char tunnel[26];
+    if(reqData.tunnelUsed){
+        snprintf(tunnel, 26, "%s%s%s", "\"", reqData.tunnel, "\"");
+    } else {
+        strcpy(tunnel,"false");
+    }
+    // Concatenate into string for post data
+    char* buf = malloc(sizeof(char) * (500));
+    snprintf(buf,
+             sizeof(char) * (500),
+             "{\"system\":\"%s\", \"os_version\":\"%s\", \"glibc_version\":\"%s\", \"hostname\":\"%s\", \"tunnel\":%s, \"ips\": [\"%s\"]}",
+             "linux",
+             reqData.osVersion,
+             reqData.glibcVersion,
+             reqData.hostname,
+             tunnel,
+             reqData.IPstring);
+    return buf;
+}
+
+void init_response(struct response *s) {
+  s->len = 0;
+  s->ptr = malloc(s->len+1);
+  if (s->ptr == NULL) {
+    fprintf(stderr, "malloc() failed\n");
+    exit(EXIT_FAILURE);
+  }
+  s->ptr[0] = '\0';
+}
+
+size_t writefunc(void *ptr, size_t size, size_t nmemb, struct response *s)
+{
+  size_t new_len = s->len + size*nmemb;
+  s->ptr = realloc(s->ptr, new_len+1);
+  if (s->ptr == NULL) {
+    fprintf(stderr, "realloc() failed\n");
+    exit(EXIT_FAILURE);
+  }
+  memcpy(s->ptr+s->len, ptr, size*nmemb);
+  s->ptr[new_len] = '\0';
+  s->len = new_len;
+
+  return size*nmemb;
+}
+
+
+char* executeCommand(char* commandLine){
+    FILE *fp;
+    char* path = (char *) malloc(sizeof(char) * 300);
+
+    /* Open the command for reading. */
+    fp = popen(commandLine, "r");
+    if (fp == NULL) {
+    printf("Failed to run command\n" );
+    exit(1);
+    }
+    /* Read the output a line at a time - output it. */
+    fgets(path, 300, fp);
```

Comment:
You don't check `feof` so we don't know if there's more input to read. You also don't check if we failed reading.